image_transport/test/test_publisher.cpp

```diff
@@ -46,31 +46,87 @@ class TestPublisher : public ::testing::Test
   rclcpp::Node::SharedPtr node_;
 };
 
-TEST_F(TestPublisher, Publisher) {
+TEST_F(TestPublisher, publisher) {
   auto pub = image_transport::create_publisher(node_.get(), "camera/image");
 }
 
-TEST_F(TestPublisher, ImageTransportPublisher) {
+TEST_F(TestPublisher, image_transport_publisher) {
   image_transport::ImageTransport it(node_);
   auto pub = it.advertise("camera/image", 1);
 }
 
-TEST_F(TestPublisher, CameraPublisher) {
+TEST_F(TestPublisher, camera_publisher) {
   auto camera_pub = image_transport::create_camera_publisher(node_.get(), "camera/image");
 }
 
-TEST_F(TestPublisher, ImageTransportCameraPublisher) {
+TEST_F(TestPublisher, image_transport_camera_publisher) {
   image_transport::ImageTransport it(node_);
   auto pub = it.advertiseCamera("camera/image", 1);
 }
 
-TEST_F(TestPublisher, Shutdown) {
+TEST_F(TestPublisher, shutdown) {
   auto pub = image_transport::create_publisher(node_.get(), "camera/image");
   EXPECT_EQ(node_->get_node_graph_interface()->count_publishers("camera/image"), 1u);
   pub.shutdown();
   EXPECT_EQ(node_->get_node_graph_interface()->count_publishers("camera/image"), 0u);
 }
 
+TEST_F(TestPublisher, qos_override) {
+  // without options
```

Comment:
You should make the with and without options separate test cases rather than recreating the initial conditions halfway through the test case.

Reply:
@gbiggs separate with and without options test case at https://github.com/ros-perception/image_common/pull/208/commits/16b9d874c3c54378c82829e5754f7c1950ebb84a